pom.xml

```diff
@@ -186,6 +186,14 @@
               <nohelp>true</nohelp>
             </configuration>
           </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <version>2.21.0</version>
```

Comment:
Latest version is `2.22.1`. Would that also work?

Reply:
Judging by the SO page, it seems the main fix is not set the SureFire plugin version, but rather to set `useSystemClassLoader` to false. Please change commit message to something like:
```
Workaround for SureFire plugin bug

Set `useSystemClassLoader` to false to avoid bug reported in
https://stackoverflow.com/questions/50661648/spring-boot-fails-to-run-maven-surefire-plugin-classnotfoundexception-org-apache
```
I hope we can remove this block once the problem has been fixed the on the OpenJDK (or Maven) side.

Reply:
Done :+1:

Reply:
If you or @thomasspriggs can create create PR for test-gen it would be great. Thanks!